test/TPP/perf/perf-ops.mlir

```diff
@@ -0,0 +1,107 @@
+// RUN: tpp-opt %s -split-input-file -canonicalize | FileCheck %s
+
+// CHECK-LABEL: @perf_matmul_bench
+func.func @perf_matmul_bench(%A: tensor<4x8xf32>,
+          %B: tensor<8x4xf32>, %C: tensor<4x4xf32>, %n: i64) {
+  // CHECK: perf.bench
+  %deltas = perf.bench (%n) {
+    // CHECK: linalg.matmul
+    %D = linalg.matmul ins(%A, %B: tensor<4x8xf32>, tensor<8x4xf32>) outs(%C: tensor<4x4xf32>) -> tensor<4x4xf32>
+    // CHECK: perf.do_not_opt
+    perf.do_not_opt(%D) : tensor<4x4xf32>
+  } -> memref<?xf64>
+
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @perf_yield
+func.func @perf_yield(%a: i32, %b: i32, %n: i64) {
+  // CHECK: perf.bench
+  %deltas = perf.bench (%n) {
+    // CHECK: arith.addi
+    %c = arith.addi %a, %b : i32
+    // CHECK: perf.do_not_opt
+    perf.do_not_opt(%c) : i32
+    perf.yield
+  } -> memref<?xf64>
+
+  return
+}
+
+// -----
+
+// CHECK-LABEL: @perf_timer
+func.func @perf_timer(%a: i32, %b: i32, %n: i64) -> i32 {
+  // CHECK: perf.start_timer
+  %t = perf.start_timer : i64
+  // CHECK: arith.addi
+  %c = arith.addi %a, %b : i32
+  // CHECK: perf.stop_timer
+  perf.stop_timer(%t : i64) : f64
+
+  return %c : i32
+}
+
+// -----
+
+// CHECK-LABEL: @perf_timer
+func.func @perf_timer(%a: i32, %b: i32, %n: i64) -> i32 {
+  // CHECK: perf.start_timer
+  %t = perf.start_timer : i64
+  // CHECK: arith.addi
+  %c = arith.addi %a, %b : i32
+  // CHECK: perf.stop_timer
+  perf.stop_timer(%t : i64) : f64
+
+  return %c : i32
+}
+
+// -----
+
+// CHECK-LABEL: @perf_mean
+func.func @perf_mean(%arg0: memref<?xf64>) -> f64 {
+  // CHECK: perf.mean
+  %mean = perf.mean(%arg0 : memref<?xf64>) : f64
+  return %mean : f64
+}
+
+// -----
+
+// CHECK-LABEL: @perf_stdev
+func.func @perf_stdev(%arg0: memref<?xf64>, %mean: f64) -> f64 {
+  // CHECK: perf.stdev
+  %stdev = perf.stdev(%arg0 : memref<?xf64>, %mean : f64) : f64
+  return %stdev : f64
+}
+
+// -----
+
+// CHECK-LABEL: @perf_matmul_loops
```

Comment:
also please add a full test with `perf.bench` alternative

Reply:
That would be the currently present `perf_matmul_bench` test, if I get you correctly.
I'll reorder the test cases so the two are together.

Reply:
Right, I mean full bench + mean/stdev, so that we have "documented" both usages with the most common one represented.

What would be helpful (at this stage) is to write the above test as returning two floats, mean/stdev, then calling `bench` + `yield` + `mean` + `stdev` and returning those. Then the one below, with a comment like "this is how it's supposed to lower to", and a literal loop + start/stop, etc.

So that when you add the lowering, we can CHECK against the structure below.

It doesn't change the implementation or the design, but it documents what's expected and what are the next steps, so make it easier for you to follow up on the work and people reviewing the patches to know what to expect and spend less time on the next review.